pdir/api.py

```diff
@@ -149,11 +149,8 @@ def _getattr(self):
         """A wrapper around getattr(), handling some exceptions."""
         attrs = {}
         for name in dir(self.obj):
-            try:
-                # Ensures we get descriptor object instead of its return value.
-                attrs[name] = get_dict_attr(self.obj, name)
-            except AttributeError:
```

Comment:
Why remove this? Not everything is in `__dict__`.

Reply:
Indeed... I neglected some cases. This should be fixed soon

Reply:
I don't quite understand what those tests are, but please go ahead.

Reply:
Actually, seems like the default implementation of `dir` is by looking up `__dict__`
https://github.com/python/cpython/blob/v3.6.4/Objects/typeobject.c#L4469-L4514

So it's probably ok to remove `getattr`. As for user-defined `__dir__`, we'll need to discuss how to deal with it, cause user can put anything(like some names that don't exist) in the return value of `__dir__`.

Reply:
I add a test with custom `__dir__` and marked it as `xfail`. I believe to deal with custom `__dir__` probably another issue/PR is required.
For the `__dict__` thing, I tried to find a reasonable example but failed. Maybe the current implementation can deal with some tricks around `__dict__`.

Reply:
Whether you want to remove `getattr` or keep it, I'm ok with both.